core/rawdb/accessors_chain.go

```diff
@@ -727,12 +727,6 @@ func ReadLogs(db ethdb.Reader, hash common.Hash, number uint64, config *params.C
 	}
 	receipts := []*receiptLogs{}
 	if err := rlp.DecodeBytes(data, &receipts); err != nil {
-		// Receipts might be in the legacy format, try decoding that.
-		// TODO: to be removed after users migrated
-		if logs := readLegacyLogs(db, hash, number, config); logs != nil {
-			return logs
-		}
-		log.Error("Invalid receipt array RLP", "hash", hash, "err", err)
```

Comment:
Can we somehow keep this error log? Instead of always return nothing (e.g. people still using the legacy-format database), the error log is much more helpful.

Reply:
Yep done.